src/ansys/units/dimensions.py

```diff
@@ -48,6 +48,14 @@ def dimensions(self):
         dims = {x.value: y for x, y in self._dimensions.items()}
         return dict(dims)
 
+    @property
+    def single_dimension(self):
```

Comment:
I'm a little on the fence about this method name.  I'm good with the PR but lets review this naming.  It just does not seem intuitive if what it does is consistent with the docstring here.

Reply:
I also think that it seems there's slightly more to this:
- isn't it the caller's assumption that the dict has precisely `{ x : 1 }` for some `x`? I think that's the error condition to check.
- I think we should be returning `x` and not `x.name`?

Reply:
I'm still a tiny bit confused why there is a function to return the dimension if there is only one but raise an error if there is more than one, but the name seems better.

Reply:
_a tiny bit confused why there is a function to return the dimension_

Yeah, me too, it helps a bit to see the usage, but nevertheless I think that it makes more sense for that code to do it all within its implementation than to pollute this interface.

Reply:
I put this in to get the "MASS", "LENGTH", "TIME" ect. instead of getting it from the type. "type" is no longer a unit necessary. The only place that uses this is `UnitSystem`. 

I have added Sean's code.

Reply:
OK, so this function is only called by `UnitSytem` right? So, at the very least it should be a private member there and does not need to be tested explicitly.  Looking at the usage in the constructor/initializer of that class it's doing some kind of validation right.  I'm still trying to understand that.

Reply:
I think we can remove this function entirely.  See comment in the initialization code for `UnitSystem`.

Reply:
*Strong argument for removing this method and implementing it as a private function in systems.py*: there's no likely reason for using it outside the implementation, so we don't really want to make it public and have to document it (not trivial: for instance, we should state that this method raises, which exception type, and why). It can easily be implemented in systems.py because we expose `dimensions()` method publicly here. Let's make a hard rule based on these criteria.

BTW: Class rename
`class FundamentalDimension(Enum)` (currently `BaseDimensions`)

Reply:
Yes Sean.  If we keep it then it should be private in `UnitSystem` which is ok, I just could not spot a reason to keep it at all.  

Not sure about your class rename point, I thought we decided to go fundamental -> base terminology everywhere.

Reply:
I dimensions.dimensions to return the full dictionary it contains.
![image](https://github.com/ansys/pyansys-units/assets/143635850/bd4b30d4-4344-4115-a49e-eb841950f5c4)
In unit system it separates the dimension name from the dimension dict.
![image](https://github.com/ansys/pyansys-units/assets/143635850/75fae8c3-938a-4a9a-a761-f83165076ffe)
Is this a better way to do it?

Reply:
Yeah, I think before I could say yes or no to that I would need to understand why this validation check is there in UnitSystem at all.  If you have a look at my other comment about the initialization I don't think we need this function at all for use over in UnitSystem.  I may be misunderstanding or misreading the code, in which case the functionality to do the validation should be implemented into UnitSystem and Unit should not be extended at all.